File: leetcode/src/main/java/com/giwankim/leetcode/CorporateFlightBookings.java
package com.giwankim.leetcode;

public class CorporateFlightBookings {
  public int[] corpFlightBookings(int[][] bookings, int n) {
    int[] result = new int[n];

    for (int[] booking : bookings) {
      int first = booking[0];
      int last = booking[1];
      int seats = booking[2];
      result[first - 1] += seats;
      if (last < n) {
        result[last] -= seats;
      }
    }

    for (int i = 1; i < result.length; i++) {
      result[i] += result[i - 1];
    }
    return result;
  }
}

File: leetcode-kt/src/main/kotlin/com/giwankim/leetcode/CorporateFlightBookings.kt
package com.giwankim.leetcode

class CorporateFlightBookings {
    fun corpFlightBookings(
        bookings: Array<IntArray>,
        n: Int,
    ): IntArray {
        val result = IntArray(n)
        for ((first, last, seats) in bookings) {
            result[first - 1] += seats
            if (last < n) {
                result[last] -= seats
            }
        }
        for (i in 1 until result.size) {
            result[i] += result[i - 1]
        }
        return result
    }
}
